### ISSUE TITLE
[$2000] Toggle switch fails to toggle back when tapped again on preferences page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#25880]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open settings
2. Open preferences
3. tap on Toggle to ON/OFF
## Expected Result: 
Toggle switch should switch to ON/OFF when it is tapped again

## Actual Result: 
The toggle switch only works once and does not respond to subsequent taps
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome - Samsung SM-A315G Android 12
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** needs reproduction
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/771bd7e9-d0ba-4312-ad36-10d25f37a9c8

**Expensify/Expensify Issue URL:**
**Issue reported by:**@jo-ui
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692040308927449

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d30a31388a44c81c</li>
        <li>Upwork Job ID: 1696550383851593728</li>
        <li>Last Price Increase: 2023-12-19</li>
    </ul>
</details>

**Comment by strepanier03** [comment]
We don't have this Samsung phone type in Browserstack so I'm stuck on repro'ing. I [raised](https://expensify.slack.com/archives/C01SKUP7QR0/p1692919909835969) this for discussion internally on a plan to move forward.

**Comment by jo-ui** [comment]
@strepanier03 It does not have to be the same phone just try it on android 12

**Comment by jo-ui** [comment]
@strepanier03 it is also reproducible on other android versions

**Comment by strepanier03** [comment]
Aaah, yeah I just realized I'm updated to Android 13 and so are our browserstack models. I am clarifying best practice and will move it forward then.

**Comment by strepanier03** [comment]
Since 68% of Android users are on Android 12 or lower, we want to fix this. Moving forward!

**Comment by strepanier03** [comment]
Waiting for proposals Melvin.

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The switches on preferences page are only working once

### What is the root cause of that problem?

In Switch component we are using PressableWithFeedback, in the onPress method of PressableWithFeedback here: https://github.com/Expensify/App/blob/d447b95f71d4ce2c2400a913e8886f817c8add68/src/components/Pressable/PressableWithFeedback.js#L76

We use the singleExecution hook which goal is to guarantee that we don't run the method twice if we press twice before the first method run hasn't finished.

In this hook here:
https://github.com/Expensify/App/blob/d447b95f71d4ce2c2400a913e8886f817c8add68/src/hooks/useSingleExecution.js#L26-L34
We set isExecuting to false in the callback of InteractionManager.runAfterInteractions

The root cause is that in our case runAfterInteractions is never called so isExecuting stays at true and we are onPress method is never called

### What changes do you think we should make in order to solve the problem?

I don't think using InteractionManager.runAfterInteractions is safe because we don't have the control of when this is called or not. Removing it resolve the issue and doesn't seem to create any problem.


Other solutions:

- We could use it conditionally by passing a prop to the Pressable since it seems runAfterInteractions is not called only in some cases, we could also only use it when we really need it with that prop
- We could use another pressable than PressableWithFeedback in Switch component
- We could use a timeout







### What alternative solutions did you explore? (Optional)

**Comment by ShogunFire** [comment]
It's hard to have a specific solution has long as we don't know in which cases runAfterInteractions was used, I will try to find the reason. 

PS: I am almost sure that I was able to reproduce on windows chrome at some point during my tests but now I can't reproduce

**Comment by strepanier03** [comment]
Thanks @ShogunFire. I'll let @thesahindia weigh in as the more knowledgeable person about the code here and continue following along.

**Comment by ShogunFire** [comment]
I don't think that depends on the android version I was reproducing on android 13 on emulator, maybe since this is on web that has to do with chrome version ?

**Comment by strepanier03** [comment]
@thesahindia, how's it going with reproing?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ShogunFire** [comment]
I still reproduce on latest main

**Comment by strepanier03** [comment]
@ShogunFire - What is your Chrome version? I have never been able to repro this and it looks like KI testing couldn't either.

**Comment by ShogunFire** [comment]
@strepanier03 
![image](https://github.com/Expensify/App/assets/19537677/e1953703-2ebb-455f-b9a1-943c8fed2cd5)

**Comment by jo-ui** [comment]
@strepanier03 here is another chrome version that this issue is reproducible![Screenshot_20230916-074749_Chrome.jpg](https://github.com/Expensify/App/assets/57154082/37e09b34-eb0d-445b-8904-d6066671b1a7)

**Comment by strepanier03** [comment]
I am on Chrome version: 116.0.5845.173 and wasn't able to repro still. 

There wasn't a more up-to-date version of Chrome to download so I am still unable to repro and at a loss of how to help someone else repro.

**Comment by ShogunFire** [comment]
Maybe it has nothing to do with chrome version then :( InteractionManager.runAfterInteractions is not called in some cases but not sure what are the conditions

**Comment by ShogunFire** [comment]
A quick workaround would be to add a timeout if the runAfterInteractions is never called but that's probably not the best way: https://github.com/facebook/react-native/issues/8624#issuecomment-231040370

**Comment by strepanier03** [comment]
I'm switching this back to External because it looks like we're making progress in the right direction.

**Comment by strepanier03** [comment]
If we should do something else, ping me and I'll take care of it.

**Comment by roryabraham** [comment]
It seems like this issue is kind of in limbo. @ShogunFire do you want to use our proposal template to propose a solution for this issue?

**Comment by shubham1206agra** [comment]
@roryabraham If this is low priority, and if @ShogunFire RCA is correct, can we wait for https://github.com/Expensify/App/issues/28132?

**Comment by strepanier03** [comment]
Friendly bump @roryabraham for [this](https://github.com/Expensify/App/issues/25880#issuecomment-1735604370) but mostly to get this out of overdue because I don't think it needs to be marked that way right now.

**Comment by ShogunFire** [comment]
@strepanier03 I think the C+ was @thesahindia, I just tagged @roryabraham because he worked on something related.

I updated my proposal with the new informations and some possible solutions but I haven't been able to test yet. I wanted to try with setDeadline but we don't have this method yet in our repo and when I link the package to the upstream repo there is some errors. What is the process to update our repo ?

I will try to make a more specific proposal because I don't think waiting for lottie update will solve it

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by strepanier03** [comment]
>  I wanted to try with setDeadline but we don't have this method yet in our repo and when I link the package to the upstream repo there is some errors. What is the process to update our repo ?


@ShogunFire - I'm not sure the answer for this one, you'll want to reach out in Slack to raise this question. Sorry I don't have the info that would help here.

**Comment by strepanier03** [comment]
We're okay Melvin. @ShogunFire is working on a more specific proposal as well.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by strepanier03** [comment]
Thank you @mvtglobally 👍

**Comment by strepanier03** [comment]
@ShogunFire - Did you still plan to submit a more specific proposal for this?

**Comment by ShogunFire** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/25880#issuecomment-1707168269)

Sorry for the delay, I have this proposal that works by changing lottie code, otherwise we can wait https://github.com/Expensify/App/issues/16660 which will remove the fork of react-native-web and allow us to use setDeadline

**Comment by strepanier03** [comment]
Got it! I'll put this on hold and move to Weekly. When we're ready to move forward we can change this back to Daily.

**Comment by strepanier03** [comment]
Still on hold, the linked PR is open and we're waiting on that.

**Comment by strepanier03** [comment]
Still holding for PR merge.

**Comment by strepanier03** [comment]
Still on hold.

**Comment by shubham1206agra** [comment]
@strepanier03 I think PR has already merged. We can safely get this off hold.

**Comment by strepanier03** [comment]
Great, I've taken this off hold and moved it back to Daily.

**Comment by strepanier03** [comment]
Friendly bump @ShogunFire, thanks!

**Comment by strepanier03** [comment]
Still waiting for proposals, all good melvin.

**Comment by strepanier03** [comment]
I'll test this again today or tomorrow.

**Comment by Tony-MK** [comment]
It seems to work on Andriod 10. 
However, there is a problem with Android 12. 

@strepanier03 Could you please retest this again?

**Comment by strepanier03** [comment]
Was out of the office. Working on this now.

**Comment by strepanier03** [comment]
Here's what I found:

- Physical device, a Pixel 6 Pro is on Android 14 - Can't reproduce
- Browserstack on a virtual Pixel 6 pro on Android 12 - Can't reproduce

@Tony-MK - Can you add a screencast of your repro and a bit more about your testing device so I can try to emulate your testing parameters?

**Comment by Tony-MK** [comment]
@strepanier03, Since you couldn't reproduce the bug on Android 12, could this other problem only apply to me? 

Listed devices from left to right respectively:

1) Pixel 6 Pro - Android 13.0
2) Pixel 7 Pro - Android 12.0
3) Pixel 7 Pro - Android 12L
4) Pixel 6 - Android 12L

### Staging
![Staging Screenshot](https://github.com/Expensify/App/assets/22982173/d2f38308-ef42-43fa-9839-a4da032c0861)

### Production
![Production Screenshot](https://github.com/Expensify/App/assets/22982173/dee9753f-d084-4338-83b3-456292cdfd32)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week)

**Comment by strepanier03** [comment]
@Tony-MK - I guess it could be specific to you, but I'm not 100% sure. 

I'll do another round of testing and if I still can't repro we can close this out until it becomes relevant again, at which point we can reopen.

**Comment by strepanier03** [comment]
### Testing results 

Android Chrome

- Physical Pixel 6 Pro, Android 14 - Chrome = No repro
- Virtual Samsung S23, Android 13 - Chrome = No repro
- Virtual Samsung S21, Android 12 - Chrome = No repro
- Virtual Samsung S21, Android 11 - Chrome = No repro
- Virtual Samsung S20 Ultra, Android 10 - Chrome = No repro

Okay, so I think at this point we close. I've tested on various Android versions and can't reproduce this any longer.

**Comment by Tony-MK** [comment]
Thanks a lot for the confirmation.
